README.md

```diff
@@ -1,2 +1,154 @@
+
+- [Background](#background)
+  * [Example DID Document](#example-did-document)
+- [Security](#security)
+- [Install](#install)
+- [Usage](#usage)
+- [Contribute](#contribute)
+- [Commercial Support](#commercial-support)
+- [License](#license)
+
+## Background
+
+See also (related specs):
+
+* [Decentralized Identifiers (DIDs) - Data Model and Syntaxes](https://w3c-ccg.github.io/did-spec/)
+* [Linked Data Cryptographic Suite Registry](https://w3c-ccg.github.io/ld-cryptosuite-registry/)
+* [Linked Data Proofs](https://w3c-dvcg.github.io/ld-proofs/)
+
+A `did:key` method driver for the [`did-io`](https://github.com/digitalbazaar/did-io)
+client library and for standalone use.
+
+The `did:key` method is used to express public keys in a way that doesn't 
+require a DID Registry of any kind. Its general format is:
+
+```
+did:key:<multibase encoded, multicodec identified, public key>
+```
+
+So, for example, the following DID would be derived from a base-58 encoded 
+ed25519 public key:
+
+```
+did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH
+```
+
+That DID would correspond to the following DID Document:
+
+### Example DID Document
+
+```json
+{
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+  "publicKey": [
+    {
+      "id": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+      "type": "Ed25519VerificationKey2018",
+      "controller": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+      "publicKeyBase58": "B12NYF8RrR3h41TDCTJojY59usg3mbtbjnFs7Eud1Y6u"
+    }
+  ],
+  "authentication": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "assertionMethod": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "capabilityDelegation": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "capabilityInvocation": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "keyAgreement": [
+    {
+      "id": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH#zBzoR5sqFgi6q3iFia8JPNfENCpi7RNSTKF7XNXX96SBY4",
+      "type": "X25519KeyAgreementKey2019",
+      "controller": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+      "publicKeyBase58": "JhNWeSVLMYccCk7iopQW4guaSJTojqpMEELgSLhKwRr"
+    }
+  ]
+}
+```
+
+## Security
+
+The `keyAgreement` key is a Curve25519 public key (suitable for 
+Diffie-Hellman key exchange) that is deterministically _derived_ from the source 
+Ed25519 key, using  [`ed2curve-js`](https://github.com/dchest/ed2curve-js). 
+
+Note that this derived key is optional -- there's currently 
+[no proof](https://crypto.stackexchange.com/questions/3260/using-same-keypair-for-diffie-hellman-and-signing/3311#3311) 
+that this is safe to do.
+
+## Install
+
+Requires Node.js 8.3+
+
+To install locally (for development):
+
+```
+git clone https://github.com/digitalbazaar/did-method-key.git
+cd did-method-key
+npm install
+```
+
```

Comment:
Could add docs for a direct npm install, assuming this will be published soon.

Reply:
k!

Reply:
Added